### DISCUSSION TITLE
Performance evaluation with OpenSSL

### DISCUSSION THREAD

**Discussion by Ashok-d-attikatti** [openssl/openssl#23020] [Q&A]
We are trying to evaluate the HCA with the OpenSSL utility. but we are observing the log as below
```
Engine "devcrypto" set.
Doing AES-128-CBC for 3s on 16 size blocks: 512457 AES-128-CBC's in 0.13s
Doing AES-128-CBC for 3s on 64 size blocks: 413807 AES-128-CBC's in 0.18s
Doing AES-128-CBC for 3s on 256 size blocks: 209666 AES-128-CBC's in 0.08s
Doing AES-128-CBC for 3s on 1024 size blocks: 70848 AES-128-CBC's in 0.01s
Doing AES-128-CBC for 3s on 8192 size blocks: 8898 AES-128-CBC's in 0.01s
Doing AES-128-CBC for 3s on 16384 size blocks: 4190 AES-128-CBC's in 0.00s
```

we see that **0.13s** for generating the block of **512457** (16 size). we would like to know if this data is generated for **3s or 0.13s.**?
we would like to know how we can get the data rate when processing the above data blocks.?

Thanks,
Ashoka A D

**Comment by t8m** [comment]
These lines are just informative. Use the summary output to compare performance. Example:
```
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes  16384 bytes
AES-128-CBC    1225675.02k  1499306.71k  1558803.97k  1575007.23k  1579813.55k  1580537.17k
```

**Reply by Ashok-d-attikatti** [reply]
Hi @t8m 
we would like to know the **1225675.02k** data is processd for **3s or 1s or 0.13s**

Thanks,
Ashoka A D

**Reply by t8m** [reply]
There is this sentence in the speed output: `The 'numbers' are in 1000s of bytes per second processed.` I.e., `1225675.02k` means 1225675020 bytes processed per 1 second.

**Reply by Ashok-d-attikatti** [reply]
Thanks for your update. we understood the point
